### PR TITLE
uavobjectparser: improvements to validation, error messages

### PR DESCRIPTION
**Pull request by mlyle**
It's less confusing now when something is wrong with a uavo xml file.

### CODE REVIEW

File: ground/uavobjgenerator/uavobjectparser.cpp [hunk 1]
```diff
@@ -117,6 +118,26 @@ bool fieldTypeLessThan(const FieldInfo* f1, const FieldInfo* f2)
...
 
 /**
+ * Returns a human-meaningful error message
+ * @param fileName The filename we found an error in
+ * @param errMsg A human readable description of what's wrong
+ * @param errorLine The line number where this problem occurred
+ * @param errorCol The column number
+ * @returns A formatted error string.
+ */
+QString UAVObjectParser::genErrorMsg(QString& fileName,
```
Comment (guilhermito) [inline ground/uavobjgenerator/uavobjectparser.cpp]: operator should be next to variable name not type

**Comment by BrainFPV** [comment]
LGTM

File: ground/uavobjgenerator/uavobjectparser.cpp (hunk 1 above)
Comment (mlyle) [inline ground/uavobjgenerator/uavobjectparser.cpp]: I agree-- I like it that way.

Unfortunately, that whole file, and all of the QT code we have puts the referencing operator next to the type name.

**Comment by mlyle** [comment]
Cool, thanks for the merges.

**Pull request merged**